repl/scala-2.10/src/main/scala/org/apache/spark/repl/SparkILoop.scala

```diff
@@ -1008,7 +1008,7 @@ class SparkILoop(
     val jars = SparkILoop.getAddedJars
     val conf = new SparkConf()
       .setMaster(getMaster())
-      .setAppName("Spark shell")
+      .setAppName(sys.props.getOrElse("spark.app.name", "Spark shell"))
```

Comment:
You probably could avoid all changes to the shell scripts and just say:

    .setIfMissing("spark.app.name", "Spark shell")